src/Shared/ReuseableStringBuilder.cs

```diff
@@ -186,6 +186,35 @@ internal ReuseableStringBuilder Append(string value, int startIndex, int count)
             return this;
         }
 
+        public ReuseableStringBuilder AppendJoined(char separator, ICollection<string> strings)
```

Comment:
`AppendSeparated`? I like the consistency with `Join` but the name feels wrong to me.